### PR TITLE
[DCA-5] Goodness from data_curator-infra repo

### PR DESCRIPTION
**Pull request by zaro0508**
This is to carry over the DCA-5 epic fixes along with other
goodies that were done in the Sage-Bionetworks/data_curator-infra
repo and apply it to this template repo.

### CODE REVIEW

File: common/vpc_stack.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import config
+
+from aws_cdk import (Stack,
+    aws_ec2 as ec2,
+    Tags)
+
+from constructs import Construct
+
+STACK_ID = "dca-common"
```
Comment (brucehoff) [inline common/vpc_stack.py]: The generic template shouldn't reference the DCA project.

**Review by brucehoff — CHANGES_REQUESTED** [review]

File: common/vpc_stack.py (hunk 1 above)
Comment (zaro0508) [reply common/vpc_stack.py]: fixed, thanks.

File: .github/workflows/aws-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: 1200
+
+        # https://github.com/marketplace/actions/aws-cdk-github-actions
+      - name: cdk synth
+        uses: youyo/aws-cdk-github-actions@v2
+        with:
+          cdk_subcommand: 'synth'
+          cdk_args: '--context env=dev --debug'
+
```
Comment (brucehoff) [inline .github/workflows/aws-deploy.yml:52]: Not sure why uncommenting code is part of the deployment process. Is it to allow testing more of the pipeline (say, in a PR against the main branch) before deploying to an AWS account (say, in another branch)?  If so, can you please add a comment to this effect in the README?

**Review by brucehoff — CHANGES_REQUESTED** [review]

File: .github/workflows/aws-deploy.yml (hunk 2 above)
Comment (zaro0508) [reply .github/workflows/aws-deploy.yml:52]: The purpose of a template repo is to just get users started.  We are not deploying this project to a real AWS account.  that's why it's commented out.  Users can uncomment and update the `role-to-assume` in this file with a real/valid role to allow deployment to a real AWS account.

Comment (brucehoff) [reply .github/workflows/aws-deploy.yml:52]: Oh, right.  My mistake. (I thought I was looking at the `data_curator-infra` repo'.)

**Review by brucehoff — APPROVED** [review]

**Pull request merged**